src/Microsoft.VisualStudio.ProjectSystem.FSharp/ProjectSystem/LanguageServices/FSharpParseBuildOptions.cs

```diff
@@ -2,19 +2,56 @@
 
 using System.Collections.Generic;
 using System.ComponentModel.Composition;
-using Microsoft.CodeAnalysis.CSharp;
+using System.IO;
+using Microsoft.CodeAnalysis;
 
 namespace Microsoft.VisualStudio.ProjectSystem.LanguageServices
 {
     [Export(typeof(IParseBuildOptions))]
     [AppliesTo(ProjectCapability.FSharp)]
     internal class FSharpParseBuildOptions : IParseBuildOptions
     {
+        private const string ReferencePrefix = "-r:";
+
         public BuildOptions Parse(IEnumerable<string> args, string baseDirectory)
         {
-            // TODO: replace with F# command line parser
-            return BuildOptions.FromCommonCommandLineArguments(
-                CSharpCommandLineParser.Default.Parse(args, baseDirectory, sdkDirectory: null, additionalReferenceDirectories: null));
+            var sourceFiles = new List<CommandLineSourceFile>();
+            var additionalFiles = new List<CommandLineSourceFile>();
+            var metadataReferences = new List<CommandLineReference>();
+
+            foreach (var arg in args)
+            {
+                if (arg.StartsWith(ReferencePrefix))
+                {
+                    // e.g., -r:C:\Path\To\FSharp.Core.dll
+                    metadataReferences.Add(new CommandLineReference(arg.Substring(ReferencePrefix.Length), MetadataReferenceProperties.Assembly));
+                }
+                else if (!arg.StartsWith("-"))
+                {
+                    // not an option, should be a regular file
+                    var extension = Path.GetExtension(arg).ToLowerInvariant();
+                    switch (extension)
+                    {
+                        case ".fs":
+                        case ".fsi":
+                        case ".fsx":
+                        case ".fsscript":
+                        case ".ml":
+                        case ".mli":
+                            sourceFiles.Add(new CommandLineSourceFile(arg, isScript: (extension == ".fsx") || (extension == ".fsscript")));
+                            break;
+                        default:
+                            additionalFiles.Add(new CommandLineSourceFile(arg, isScript: false));
```

Comment:
I don't think this is right - the F# LS doesn't understand additional files yet since there's no analyzers support. In C#\VB, additionalfiles are specific item types. I think it'd be best to make this always empty until real analyzers support is implemented.